apis/Google.Cloud.PubSub.V1/docs/index.md

```diff
@@ -82,6 +82,80 @@ restart listening for messages with `StartAsync(...)` again. Due to the expense
+
+{{sample:PublisherClient.AddPublisherClient}}
+
+There is an overload of `AddPublisherClient` method that takes `Action<PublisherClientBuilder>` as a parameter
+and can be used to add the customized `PublisherClient` singleton instance as shown below:
+
+{{sample:PublisherClient.AddCustomizedPublisherClient}}
+
+Using the `PublisherClient` registered with the dependency injection container is easy. To do so, simply create a constructor for the class in which the `PublisherClient`
+is required to be used and add the `PublisherClient` as a parameter.
+
+For example:
+
+{{sample:PublisherClient.UsePublisherClient}}
+
+When the application exits, the `DisposeAsync` method of the `PublisherClient` will be invoked automatically to gracefully shut down the client.
+
+### SubscriberClient
+
+To register a singleton `SubscriberClient` instance with default settings in the `IServiceCollection`, use the
+`AddSubscriberClient` extension method as shown below:
+
+{{sample:SubscriberClient.AddSubscriberClient}}
+
+There is an overload of `AddSubscriberClient` method that takes `Action<SubscriberClientBuilder>` as a parameter
+and can be used to add the customized `SubscriberClient` singleton instance as shown below:
+
+{{sample:SubscriberClient.AddCustomizedSubscriberClient}}
+
+The `SubscriberClient` is a long-running client and so it may be useful to use
+it in a background service. The background service can use the `SubscriberClient`
+registered with the dependency injection container and handle the messages in the background.
+
+In ASP.NET Core, the background services can be registered with the dependency injection container
+using the [`AddHostedService`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.servicecollectionhostedserviceextensions.addhostedservice?view=dotnet-plat-ext-6.0) extension method as shown below:
+
+{{sample:SubscriberClient.AddHostedService}}
+
+Here `SubscriberService` is the class that implements `BackgroundService` and uses the `SubscriberClient`
+registered with the dependency injection container to handle the messages. Once the service is registered,
+it will be automatically started when the application starts and stopped when the application exits.
+A sample implementation of `SubscriberService` is shown below:
+
+{{sample:SubscriberClient.UseSubscriberClient}}
+
+During application shutdown, the `StopAsync` method of the `SubscriberService` is invoked, which in turn calls
+the `StopAsync` method of the `SubscriberClient` to gracefully shut down the client.
+
+## IAsyncDisposable
+
+Both `PublisherClient` and `SubscriberClient` implement the `IAsyncDisposable` interface,
+allowing for asynchronous resource cleanup.
+
+For the `PublisherClient`, when the `DisposeAsync` method is called, it asynchronously waits for the time interval
+specified in the `PublisherClient.Settings.DisposeTimeout` property for the `PublisherClient` to send any pending messages before aborting
```

Comment:
Will this work well with DI? What happens if I set this to 100 hours for instance? Will DI just wait for it to be done or should I also configure DI to wait for that long?